src/scanner.c

```diff
@@ -0,0 +1,403 @@
+
+#include <tree_sitter/parser.h>
+#include <ctype.h> // isspace()
+#include <string.h> // memcpy()
+
+enum TokenType {
+  NEWLINE,
+  SEMICOLON,
+  ELSE,
+  RAW_STRING_LITERAL,
+  OPEN_PAREN,
+  CLOSE_PAREN,
+  OPEN_BRACE,
+  CLOSE_BRACE,
+  OPEN_BRACKET,
+  CLOSE_BRACKET,
+  OPEN_BRACKET2,
+  CLOSE_BRACKET2
+};
+
+// ---------------------------------------------------------------------------------------
+// Temporary Stack structure until we can use the `<tree_sitter/array.h>` header from
+// tree sitter 1.0.0. Inspired from tree-sitter-julia.
+
+typedef char Scope;
```

Comment:
Particularly important to use `char` as the element storage, as it makes the `malloc()` and `memcpy()` calls simple (no sizeof needed).

For this reason I didn't introduce an `enum Scope` because I think that enum values are `int`s